internal/support/notifications/v2/application/subscription.go

```diff
@@ -155,12 +155,11 @@ func PatchSubscription(ctx context.Context, dto dtos.UpdateSubscription, dic *di
 
 	requests.ReplaceSubscriptionModelFieldsWithDTO(&subscription, dto)
 
-	edgexErr = dbClient.DeleteSubscriptionByName(subscription.Name)
-	if edgexErr != nil {
-		return errors.NewCommonEdgeXWrapper(edgexErr)
+	if len(subscription.Categories) == 0 && len(subscription.Labels) == 0 {
```

Comment:
Isn't this now covered by https://github.com/edgexfoundry/go-mod-core-contracts/pull/570

Reply:
This is used to prevent user updating the categories and labels to both empty. 
For example, 
- at first, user add a subscription with categories 
  ```
  {
      "apiVersion":"v2",
      "subscription":{
         "name":"Subscription-1616666801751",
         "channels":[
            {
               ...
            }
         ],
         "categories":[
            "health-check"
         ],
      }
   }
  ``` 
- at next time, user patch the subscription with empty categories .
  ```
  {
      "apiVersion":"v2",
      "subscription":{
         "name":"Subscription-1616666801751",
         "channels":[
            {
               ...
            }
         ],
         "categories":[],
      }
   }
  ``` 
This case will cause the categories and labels both empty in the DB. So I check the subscription again before updating to the DB.

```
func PatchSubscription(ctx context.Context, dto dtos.UpdateSubscription, dic *di.Container) errors.EdgeX {
	...
        // Read subscription from DB
	if dto.Id != nil {
		subscription, edgexErr = dbClient.SubscriptionById(*dto.Id)
		if edgexErr != nil {
			return errors.NewCommonEdgeXWrapper(edgexErr)
		}
	...
        // Set the DTO value to subscription
	requests.ReplaceSubscriptionModelFieldsWithDTO(&subscription, dto)

        // Ensure that the subscription's Categories and Labels can not be both empty
	if len(subscription.Categories) == 0 && len(subscription.Labels) == 0 {
		return errors.NewCommonEdgeX(errors.KindContractInvalid, "subscription categories and labels can not be both empty", nil)
	}

	edgexErr = dbClient.UpdateSubscription(subscription)
	...
	return nil
}
```

Reply:
Ok, got it. THX!